File: CS/RefreshDataExample/GridTimerRefreshBehavior.cs
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Grid;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Threading;

namespace RefreshDataExample
{
    public class GridTimerRefreshBehavior : Behavior<GridControl>
    {
        DispatcherTimer refreshTimer = new DispatcherTimer();
        protected override void OnAttached() {
            base.OnAttached();
            refreshTimer.Interval = new TimeSpan(0, 0, 0, 0, 1000);
            refreshTimer.Tick += RefreshTimer_Tick;
            refreshTimer.Start();
        }

        private void RefreshTimer_Tick(object sender, EventArgs e) {
            AssociatedObject.RefreshData();
        }
    }
}

File: VB/RefreshDataExample/GridTimerRefreshBehavior.vb
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.Grid
Imports System
Imports System.Windows.Threading

Namespace RefreshDataExample

    Public Class GridTimerRefreshBehavior
        Inherits Behavior(Of GridControl)

        Private refreshTimer As DispatcherTimer = New DispatcherTimer()

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            refreshTimer.Interval = New TimeSpan(0, 0, 0, 0, 1000)
            AddHandler refreshTimer.Tick, AddressOf RefreshTimer_Tick
            refreshTimer.Start()
        End Sub

        Private Sub RefreshTimer_Tick(ByVal sender As Object, ByVal e As EventArgs)
            AssociatedObject.RefreshData()
        End Sub
    End Class
End Namespace
